tests/EndToEndTests.hs

```diff
@@ -152,3 +155,137 @@ tests = testSpec "End-to-end tests" $ do
               ]
             ]
       toJSON (toValue response) `shouldBe` expected
+    it "It aliases fields" $ do
+      let root = pure (viewServerDog mortgage)
+      let query = [r|{
+                      dog {
+                        name
+                        boss: owner {
+                          name
+                        }
+                      }
+                    }
+                   |]
+      response <- interpretAnonymousQuery @QueryRoot root query
+      let expected =
+            object
+            [ "data" .= object
+              [ "dog" .= object
+                [ "name" .= ("Mortgage" :: Text)
+                , "boss" .= object
+                  [ "name" .= ("jml" :: Text)
+                  ]
+                ]
+              ]
+            ]
+      toJSON (toValue response) `shouldBe` expected
+    it "Passes arguments to functions" $ do
+      let root = pure (viewServerDog mortgage)
+      let query = [r|{
+                      dog {
+                        name
+                        doesKnowCommand(dogCommand: Sit)
+                      }
+                     }
+                    |]
+      response <- interpretAnonymousQuery @QueryRoot root query
+      let expected =
+            object
+            [ "data" .= object
+              [ "dog" .= object
+                [ "name" .= ("Mortgage" :: Text)
+                , "doesKnowCommand" .= False
+                ]
+              ]
+            ]
+      toJSON (toValue response) `shouldBe` expected
+  describe "interpretQuery" $ do
+    it "Handles the simplest named query" $ do
+      let root = pure (viewServerDog mortgage)
+      let query = [r|query myQuery {
+                      dog {
+                        name
+                      }
+                    }
+                   |]
+      response <- interpretQuery @QueryRoot root query Nothing mempty
+      let expected =
+            object
+            [ "data" .= object
+              [ "dog" .= object
+                [ "name" .= ("Mortgage" :: Text)
+                ]
+              ]
+            ]
+      toJSON (toValue response) `shouldBe` expected
+    it "Allows calling query by name" $ do
+      let root = pure (viewServerDog mortgage)
+      let query = [r|query myQuery {
+                      dog {
+                        name
+                      }
+                    }
+                   |]
+      let Right name = makeName "myQuery"
+      response <- interpretQuery @QueryRoot root query (Just name) mempty
+      let expected =
+            object
+            [ "data" .= object
+              [ "dog" .= object
+                [ "name" .= ("Mortgage" :: Text)
+                ]
+              ]
+            ]
+      toJSON (toValue response) `shouldBe` expected
+    describe "Handles variables" $ do
+      let root = pure (viewServerDog mortgage)
+      let Right query =
+            compileQuery [r|query myQuery($whichCommand: DogCommand) {
```

Comment:
No, it's not. That's https://facebook.github.io/graphql/#sec-All-Variable-Usages-are-Allowed, which is listed as being out of scope for the `Validation` module.

I think we could & should bring it back into scope for later releases. 

Tangentially, I think it's interesting to know what typeless GraphQL validation/compilation looks like—I wonder if the GraphQL authors even know.